homeassistant/components/broadlink/climate.py

```diff
@@ -0,0 +1,117 @@
+"""Support for Broadlink climate devices."""
+from homeassistant.components.climate import ClimateEntity
+from homeassistant.components.climate.const import (
+    CURRENT_HVAC_HEAT,
+    CURRENT_HVAC_IDLE,
+    CURRENT_HVAC_OFF,
+    HVAC_MODE_AUTO,
+    HVAC_MODE_HEAT,
+    HVAC_MODE_OFF,
+    SUPPORT_TARGET_TEMPERATURE,
+)
+from homeassistant.const import ATTR_TEMPERATURE, TEMP_CELSIUS
+from homeassistant.core import callback
+from homeassistant.helpers.restore_state import RestoreEntity
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Broadlink climate entities."""
+    device = hass.data[DOMAIN].devices[config_entry.entry_id]
+
+    if device.api.type in {"Hysen heating controller"}:
+        climate_entities = [BroadlinkHysen(device)]
+    async_add_entities(climate_entities)
+
+
+class BroadlinkHysen(ClimateEntity, RestoreEntity):
```

Comment:
This class should be renamed to BroadlinkThermostat, as the same PID is related to devices of different brands. See https://github.com/mjg59/python-broadlink/pull/500.